src/ucp/core/ucp_ep.c

```diff
@@ -3222,8 +3213,7 @@ ucs_status_t ucp_ep_do_uct_ep_keepalive(ucp_ep_h ucp_ep, uct_ep_h uct_ep,
     keepalive_type = ucp_ep_keepalive_type(ucp_ep, rsc_idx);
     if (keepalive_type == UCP_EP_KEEPALIVE_EP_CHECK) {
         return uct_ep_check(uct_ep, flags, comp);
-    } else if ((keepalive_type == UCP_EP_KEEPALIVE_BUILT_IN) ||
-               (keepalive_type == UCP_EP_KEEPALIVE_NONE)) {
+    } else if (keepalive_type == UCP_EP_KEEPALIVE_NONE) {
```

Comment:
I don't think we need to define an enum for keepalive_type
if ucp_ep_do_uct_ep_keepalive is called, it should mean keepalive_lane != NULL_LANE (because we have uct_ep parameter)
then, there would be only 2 options: am or ep_check
so ucp_ep_is_am_keepalive() should return 1 if am and 0 if ep_check

Reply:
removed